packages/babel-traverse/src/path/modification.js

```diff
@@ -112,7 +116,10 @@ export function insertAfter(nodes) {
     this.parentPath.isExpressionStatement() ||
     this.parentPath.isLabeledStatement()
   ) {
-    return this.parentPath.insertAfter(nodes);
+    // `replaceWithMultiple` requeues if there's a replacement for this.node,
+    // but not for an ancestor's. Set `shouldRequeue` to true so that any replacement
+    // for an ancestor node can be enqueued. Fix #5628 and #5023.
+    return this.parentPath._insertAfter(nodes, true);
```

Comment:
@jridgewell I propose limiting the scope of this PR to fix #5628, i.e. not transformed in destructuring. For other potential bugs/use cases, maybe we should look at in a different PR. What do you think?